src/overlays/actors/ovl_En_Look_Nuts/z_en_look_nuts.h

```diff
@@ -2,16 +2,44 @@
 #define Z_EN_LOOK_NUTS_H
 
 #include "global.h"
+#include "objects/object_dnk/object_dnk.h"
 
 struct EnLookNuts;
 
 typedef void (*EnLookNutsActionFunc)(struct EnLookNuts*, GlobalContext*);
 
+#define LOOKNUTS_GET_SPAWN_INDEX(thisx) (((thisx)->params >> 0xC) & 0xF)
+#define LOOKNUTS_GET_SCENE_FLAG(thisx) ((thisx)->params & 0x7F)
+#define LOOKNUTS_GET_PATROL_LOCATION(thisx) (((thisx)->params >> 0x7) & 0x1F)
+
+typedef enum {
+    /* 0x01 */ PATROLLING_STATE,
+    /* 0x01 */ WAITING_STATE,
+    /* 0x02 */ RUNNING_TO_PLAYER_STATE,
+    /* 0x03 */ CAUGHT_PLAYER_STATE,
+} PalaceGuardState;
```

Comment:
I think it makes most sense for all enums/typedefs to be in the headers. On top of that, most all other usages of state enums that I have seen have been in the headers for OoT and MM combined.

Reply:
This is current (and future) policy, though: https://github.com/zeldaret/mm/blob/master/CONTRIBUTING.md#naming-scheme
I suppose it is conceivable that other files could use this to check what this actor is doing, but it's very rare, and since C does not have namespaces, it's better to avoid putting stuff in headers unless explicitly required.